src/power.jl

```diff
@@ -164,6 +168,20 @@ function _eval_intersect(pow::IntervalMatrixPower)
     return intersect(_eval_multiply(pow), _eval_power(pow))
 end
 
+function _eval_sqrt(pow::IntervalMatrixPower; algorithm::String="power")
+    # decompose k = a² + b with a, b being integers
+    k = pow.k
+    a = floor(Int, sqrt(k))
+    b = k - a^2
+
+    # recursively compute M^a and M^b
+    Mᵏ = square(get(IntervalMatrixPower(pow.M, a; algorithm=algorithm)))
```

Comment:
Is line 178 the same as `get(IntervalMatrixPower(pow.M, a*a)` ? (Also, this is the functionality that i was asking in [this comment](https://github.com/JuliaReach/IntervalMatrices.jl/pull/132#discussion_r377634464). I'll create an issue).

Reply:
My question still is why not use `get(IntervalMatrixPower(pow.M, a*a; algorithm=algorithm)))` here.

Reply:
> why not use `get(IntervalMatrixPower(pow.M, a*a; algorithm=algorithm)))` here.

Because is not equivalent. We do not use `square` for arbitrary square numbers but only for powers of 2.

Reply:
Oh, damn, your version was the correct one:
`A^(a²) = A^(a*a) = (A^a)^a != (A^a)²` for `a != 2`